airbyte-integrations/connectors/source-apple-search-ads/manifest.yaml

```diff
@@ -0,0 +1,1135 @@
+        - type: AddFields
+          fields:
+            - path:
+                - date
+              value: "{{ stream_slice.start_time }}"
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/keywords_report_daily"
+  base_requester:
+    type: HttpRequester
+    url_base: https://api.searchads.apple.com/api/v4
+    authenticator:
+      type: OAuthAuthenticator
+      client_id: "{{ config.client_id }}"
+      client_secret: "{{ config.client_secret }}"
+      token_refresh_endpoint: >-
+        https://appleid.apple.com/auth/oauth2/token?grant_type=client_credentials&scope=searchadsorg
+
+streams:
+  - $ref: "#/definitions/streams/campaigns"
+  - $ref: "#/definitions/streams/adgroups"
+  - $ref: "#/definitions/streams/keywords"
+  - $ref: "#/definitions/streams/campaigns_report_daily"
+  - $ref: "#/definitions/streams/adgroups_report_daily"
+  - $ref: "#/definitions/streams/keywords_report_daily"
+
+spec:
+  type: Spec
+  connection_specification:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    required:
+      - org_id
+      - client_id
+      - start_date
+      - client_secret
+    properties:
+      org_id:
+        type: integer
+        description: >-
+          The identifier of the organization that owns the campaign. Your Org Id
+          is the same as your account in the Apple Search Ads UI.
+        title: Org Id
+        order: 0
+      end_date:
+        type: string
+        description: Data is retrieved until that date (included)
+        title: End Date
+        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}$
+        examples:
+          - "2021-01-01"
+        order: 1
+      client_id:
+        type: string
+        description: >-
+          A user identifier for the token request. See <a
+          href="https://developer.apple.com/documentation/apple_search_ads/implementing_oauth_for_the_apple_search_ads_api">here</a>
+        title: Client Id
+        airbyte_secret: true
+        order: 2
+      start_date:
+        type: string
+        description: Start getting data from that date.
+        title: Start Date
+        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}$
+        examples:
+          - "2020-01-01"
+        order: 3
+      client_secret:
+        type: string
+        description: >-
+          A string that authenticates the user’s setup request. See <a
+          href="https://developer.apple.com/documentation/apple_search_ads/implementing_oauth_for_the_apple_search_ads_api">here</a>
+        title: Client Secret
+        airbyte_secret: true
+        order: 4
+    additionalProperties: true
+
+metadata:
+  autoImportSchema:
+    campaigns: false
+    adgroups: false
+    keywords: false
+    campaigns_report_daily: false
+    adgroups_report_daily: false
+    keywords_report_daily: false
+  yamlComponents:
+    streams:
+      keywords_report_daily:
+        - errorHandler
+    global:
+      - authenticator
+  testedStreams:
```

Comment:
@topefolorunso did you actually have test data to verify these? Not copy and pastedfrom another connector, right? ;)

Reply:
No not tested yet. Still waiting on @ChristoGrab for creds

Reply:
Looking into this now. We have an existing account in LastPass, but no Oauth creds. [Working through the steps to generate them].(https://developer.apple.com/documentation/apple_search_ads/implementing_oauth_for_the_apple_search_ads_api)